Changed balance to reward leveling up skills more

onpon4 opened the pull request:
This basically is to address #835 by making skills level up more rewarding on the mage class, it makes fire do a lot more extra damage and makes all of them get more damage as they level up alongside the MP drain increase. I tested a little bit and I think it seems good so far though the numbers might need some adjustment, also it could be argued that this isn't the right way to fix that problem and maybe a different solution would be better.

Will be playing with these changes for awhile to get some more testing in.

🕷️

onpon4 commented:
Converting this to a draft for now while we continue to test this. (We should have posted it as such in the first place, sorry about that.)

I've just pushed some further changes to this which include giving Freeze the same damage as Burn, reverting the changes Layla made to slowdown amounts, making the Shock spell have a much lower crit bonus than Thunderstrike, and making the splash attacks not require the missile attacks at all (Layla previously made them only require the first level corresponding missile spell).

🦇

🦇

onpon4 commented:
So we expanded this PR to cover more ground since the problem of upgrades that aren't really worth it wasn't exclusive to mages. So here's a summary of what we have here:

* We went through and changed the way a lot of abilities get upgraded so that they either don't get such a big price hike, or get requisite damage increase / significant effect benefit, or both. Abilities changed include: all the mage elemental powers, all the ranged attack abilities, Warcry, and Cleave.
* We got rid of skills requiring other skills, it's not a bad concept in general but they basically double skill point cost and so when you get upgrade involved it makes the upgrade never worth it unless the upgrade is super broken compared to upgrades that can be had for just one skill point.
* We made sure things that were upgrade requirements for other skills were worth getting in their own right. This was already true for mage skills but we gave special attention for the others: we made Throw Knife faster than Throw Axe but widened the damage gap, and we made caltrops free but made bear traps give a longer immobilization effect that gets longer as you level it up and with much more damage.
* We also tweaked some base damage and MP costs: AoE spells cost the same as same-level missile spells, Thunderbolt does less damage, Burn and Freeze do more damage (same as ice and shock missiles), and Cleave costs less (same as Blood Strike).

🕷️

I think this seems like a good solution to making the upgrades more worth it, we're testing and so far it's been way better this

onpon4 commented:
> With equal base damage/MP costs, there's very little reason not to use the AoE spells over their projectile counterparts. I would either keep the MP cost high or keep the base damage low.

To clarify, only MP is made the same; damage is made lower, with the one exception of the Freeze spell. I think this works well for the following reasons:

1. Since we set the damage of Burn and Thundershock significantly lower than their projectile counterparts, what we believe (and tests are bearing this out) is that setting it this way makes the projectiles ideal for long-range or single enemies, while the AoE attacks are ideal for clusters of enemies.
2. All the AoE attacks have a much shorter range than the projectile counterparts.
3. Freeze, in particular, has such a short range and moves so slowly that it's really only usable as a defensive pseudo-melee attack. When it inflicts less damage and/or costs more, my experience is that it's absolutely useless and is not even worth the skill point. Making it have equivalent damage and MP cost to its projectile counterpart seems with my testing to work quite well; I'm still mostly using ice projectiles in my thunder mage setup simply on account of the range difference and the slow movement, since attempting to use Freeze offensively creates a tremendous risk of aggroing many enemies at once and losing a lot of HP.

It's possible that the AoE attacks could be made to be worth an extra cost instead, though I'm doubtful of that in the case of Freeze unless it's made to inflict so much damage so as to be pretty much broken (particularly for upgrades).

> MP costs in general might be lowered too much. Particularly for mage spells, their investment in the Mental stat should offset the costs by itself.

For mage, that's certainly possible. I don't think it's really the case for the other classes, but I can see a case to be made to instead increase damage much more with level ups on mage skills. We shied away from this simply out of concern that doing so could make the mage a broken class. There's also effectively a ceiling to where this would work: if, for example, Thunderstrike is just able to one-shot everything, additional damage beyond that would be meaningless.

Perhaps some middle-ground can be sought after: slightly more damage for slightly more MP cost.

What I will say is that because of upgrades not being worth it, I've always played the game keeping the level 1 spells, with the exception of the shield spell. Consider Thunderstrike, for example: all of the upgrades do the same damage, but with just a slight damage increase. That simply was never worth it for the cost of losing out on being basically set as far as mana was concerned.

> I think I would still prefer locking the higher tier spells behind at least one level of their "lesser" spell. i.e. You should be able to cast a small fireball before learning how to cast a huge explosion.

That sounds fine to me. That was the change Layla originally made and I can add that back in.

> Caltrops having no MP cost at all seems weird. Maybe give it a small MP cost that barely increases (or doesn't at all) for each level?

Sure, a 1 MP cost that doesn't increase would probably be fine. I'll throw that in.

> For lightning spells, maybe a small stun chance would be better than a damage buff? That would differentiate them from the fire spells, while still giving a reason to invest in them outside of ice spell synergy.

That sounds like a good idea, actually. I think the Mez thunder spell and the alpha demo thunder spell do that, don't they? I'll see about adding that, perhaps with a gradual increase to the stun chance and stun time as you upgrade it.

I do think that the damage amount should still increase, though. Maybe it could increase less, but not increasing it at all would mean having to use more attacks (and thus more MP) for the same damage of equal-level ice bolts.

🦇

onpon4 commented:
So we made some of those changes:

* Power prerequisites back, but just the level 1 ones
* Caltrops cost 1 MP statically
* AoE attacks made to cost more, not as much as previously but they cost twice as much as the corresponding missile attacks, except Freeze which we just made slightly more expensive
* Gave thunder attacks a chance of stun, chance increases with level up, but they have damage that doesn't increase very much (both go from 100 to 120)
* Made the AoE attacks have things that make them better than the missile attacks: Freeze and Burn do more damage than ice bolt and fireball, and thunderbolt has higher stun chance and longer stun length

So something we observed with the lightning stun thing is it's synergetic with the crit bonus, which is really cool and it means they don't depend on ice to be meaningful anymore.

Continuing to test. :3 I have a fire mage build and Eris has a thunder mage build and we'll also wanna test other builds too.

🕷️

onpon4 commented:
Since we're stuck with a power outage and insomnia, I suppose I'll give an update on testing.

I'm very happy with where this leaves the builds we've tested so far. Adding in that stun effect makes a thunder mage build an absolute blast to play, and the fire mage build seems to work very well, too. I also tested a ranger build with level 5 piercing shot, multi shot, and the fire spitter thing (ember shot I think?) and this build also felt great. The upgrades feel much more meaningful now and the MP cost increases feel appropriate.

We'll want to test a few more builds:

* A defensive ice mage build
* A few warrior builds, particularly a ranged warrior build
* A ranger build that uses traps a lot

Will continue to test when the power comes back and post the results. 🙂

🦇

onpon4 commented:
OK so I've hit a bit of a snag in testing when it comes to the trap moves.

Finished testing fire spells and the warrior stuffs and that all seems to work great! Had to make fire a bit stronger to compensate for lack of status effects and stuffs and had to fix an accidental change in the throw knife skill but now I'm happy with those. Cleave and Blood Strike are now equally useful and upgrading Warcry is now very worth it. Same for most of the ranged skills, did a test with maxed out piercing, multi, and ember shots and everything seemed to be well-tailored (multi is an expensive all-rounder, piercing is great for lines of enemies, and ember is great defensively).

I'm not sure what to go for with the traps though, maybe I'm just not using them right but they don't seem to be very useful like, in general, because they rely on luring enemies to them and plus each only hits one enemy and you can't shoot normal (free) shots while setting them and there's a pretty short time limit too. Is nothing new, we've never used the trap weapons for this reason, but I assume if they're there they must be useful for something (or at least intended to be useful for something) right? So like, could we get some advice on what niche the trap weapons are supposed to fill and how they're meant to be used? 😅 That way I can test that stuffs out better

🕷️

onpon4 commented:
I'll need to test this more, but I've pushed an idea for how to make the traps:

1. Made both caltrops and bear trap last for 30 seconds instead of 10 seconds
2. Made caltrops multi-hit
3. Caltrops get more of an increase in damage and an increase of slow time per level, but also an increase of MP cost from 2 to 6
4. Made bear traps stun instead of immobilize (differentiating them from caltrops more; they do a much more powerful effect for longer but are single-hit)

It's possible this might make the traps a bit broken, though then again it's possible that might be a good thing considering how awkward traps are to use, unless there's some way to use them I hadn't considered. I also don't know yet if that guaranteed stun makes bosses cheesable, I know the final boss is resistant to status effects but I think the earlier bosses aren't? Something I'll have to check, if it does make the bosses cheesable might have to go back to the drawing board for the bear trap. I don't think immobilize alone can be enough unless it's strictly an upgrade to caltrops and that wouldn't be the case with making the caltrops multi-hit, which is an idea I really like. Maybe something else would work though, like immobilize + DOT or immobilize + some other effect or maybe even immobilize + a chance of stun rather than guaranteed stun. Will have to think on it.

🕷️

onpon4 commented:
Okey, I think I might have come up with something, need to test it but here's what I came up with:

* Traps last 60 seconds
* Caltrops is multi-hit, bear trap is single-hit but more damage with both immobilize and stun
* Both traps are just 1 MP regardless of upgrade level and also instant so that they don't slow down shooting

I think it might be that last one that might make them good, this way laying traps is an added bonus on top of basic attack or even special attack, at least in theory, will continue test of course.

🕷️

onpon4 commented:
I think I am satisfied with the results of testing now, it seems to be really nice. 😄

The traps feel way better than before, still not the play style I would go for but making them instant and cheap really does wonders for them, makes that play style of setting traps work a lot better. And am very very happy with everything else.

So here is summary of all of the changes this PR does now:

* Bear Trap: Lasts for 60 seconds instead of 10, costs 1 MP at all levels now, stuns in addition to immobilizing, is instant instead of swing animation, and immobilize/stun period is longer and increases on level up (level 1 is 5 seconds, level 5 is 18 seconds), also does more damage and damage per level increments by 15 pp instead of 5 pp (level 1 damage is 120%, level 5 damage is 180%)
* Caltrops: Lasts for 60 seconds instead of 10, costs 1 MP at all levels now, is now multi-hit, does less damage at first but then damage increments by 15 pp each time instead of just 5 pp (level 1 damage is 60%, level 5 damage is 120%).
* Ember Shot level-ups now increase damage by 3 pp each level instead of 2 pp (level 5 damage is 24%).
* Multishot level-ups now increase damage by 15 pp instead of 5 pp (level 5 damage is 110%), also got rid of the massive 4 MP cost increase between level 3 and 4 (so now level 4 costs 9 MP and level 5 costs 11 MP).
* Piercing Shot level-ups now increase damage by 15 pp instead of 5 pp (level 5 damage is 130%).
* Rapid-Fire level-ups now increase MP cost by 1 instead of 3 (level 5 cost is 7 MP)
* Throw Axe now has a 400 ms cooldown, damage increased to 130% at level 1 with 30 pp increase per level (level 5 damage is 250%), and MP cost increased 4 at level 1 and grows by 2 each level (level 5 cost is 12 MP).
* Throw Knife is faster while keeping the same range (life was adjusted down to keep the range the same), also does less damage at first but increases by 15 pp as it levels up instead of 5 pp (level 1 damage is 70%, level 5 damage is 130%).
* Fireball does 130% damage at level 1 instead of 105%, and damage increases by 30 pp per level (level 5 damage is 250%).
* Ice bolt damage increases by 15 pp per level (level 5 damage is 160%).
* Shock increases damage by 5 pp per level (level 5 damage is 120%), crit bonus increased (level 1 crit chance is 75%, level 5 is 95%; this is the same as Thunderstrike), and now has a chance to stun which increases in duration and chance at level up (level 1 has a 10% chance to stun for 600 ms, level 5 has a 50% chance to stun for 1 second)
* Burn costs less (level 1 costs 4 MP, level 5 costs 12 MP), also does more damage which increases by 40 pp each level up (level 1 damage is 140%, level 5 damage is 300%).
* Freeze costs much less and cost increases by 1 per level instead of 2 (level 1 costs 3 MP, level 5 costs 7 MP), damage is higher and increases by 20 pp per level (level 1 damage is 120%, level 5 damage is 200%), and slow effect slows down more on level up rather than only increasing duration (level 5 slowdown is to 5%).
* Thunderstrike costs less (level 1 costs 4 MP, level 5 costs 12 MP), damage increases by 5 pp per level (level 5 damage is 120%), and now has a chance to stun that increases in duration and chance on level up (level 1 has a 20% chance to stun for 1 second, level 5 has an 80% chance to stun for 2 seconds).
* Warcry increases immunity duration by 3 seconds each level up instead of decreasing cooldown (level 5 duration is 22 seconds). Cooldown is a static 20 seconds at all levels.
* Cleave MP cost reduced and increases slower (level 1 costs 2 MP, level 5 costs 6 MP).
* Powers that require other powers as prerequisite only require the first level and then can be upgraded without upgrading the prerequisite power.
* Bear Trap no longer has Caltrops as a prerequisite (since Caltrops is no longer considered weaker than bear trap, just for different purpose).

Let us know what you think! :3

🕷️

onpon4 commented:
Hey, it's been awhile since we finished this, any chance of a review @dorkster?

🕷️

dorkster commented:
My apologies for not noticing this was done! I didn't see the edits to the last comment.

I'll spend some time play testing it this week. The list of changes looks good overall. Still not sure if reduced MP costs are necessary, but I'll hold that thought until I'm done testing.

onpon4 commented:
No worries! I wondered if that might be why. Thanks for testing! 🙂

Regarding reduced MP costs, I want to explain the ones that we feel in particular are absolutely essential unless some other significant change is made:

* Caltrops and Bear Trap: Lowering the cost of these is necessary to make it possible to use them less efficiently without having to worry about MP. Traps have that inherent disadvantage of enemies needing to happen upon the trap, which usually doesn't happen especially considering they're used by a ranged fighter with free long-range attacks. Making them both instant and cheap allows a trap-using player to use the traps without having to worry that only a few of them hit an enemy. In short, the lower MP cost of traps means that they're designed with the expectation that most of them will fail to catch any targets, an expectation that bore out as expected in my testing.
* Freeze: This skill has a lot going against it in the magician skill tree. Its nominal benefits are that it hits multiple targets and tremendously slows enemies down, but both of these are hindered by the fact that it's short-range and slow. Because it's short-range and directed, it's not likely that you'll hit more than a couple of targets, and you also can't use it offensively without a tremendous risk to your HP (and this is amplified by the fact that fire enemies, which it is theoretically strong against, either have a tendency to keep their distance or are dangerous to get close to). While it slows down enemies more than ice bolts, in practice, there's a diminishing return on slowdown that makes Freeze's slowdown not really any more effective than the 50% slowdown of ice bolts in most cases, even with the slowdown buff we applied (which makes level 5 Freeze slow enemies down to 5%). This is simply because if you're using ice moves defensively in order to get away from them, making your speed faster than theirs is the most important, and 50% is, in practice, generally plenty to achieve this. The long and short of it is that unless Freeze does even more damage than Burn, or some other special effect is added to Freeze, it just isn't worth a high MP cost. Making Freeze do more damage than Burn wouldn't make much thematic sense and would render its slowdown redundant, and we couldn't come up with some other status effect that would complement the slowdown, so we decided to just lower the MP cost, which makes Freeze at least viable as a defensive skill. If you can think of a way to make Freeze more viable as a high-cost skill with some sort of complementary debuff or something, do let us know! I would certainly love for it to be really good skill like Burn and Thunderstrike are rather than just ok.
* Cleave: This skill is in a really awkward place of competing with a basic attack that does about as much damage as it, but doesn't cost any MP. We felt that, given its damage output, costing almost twice as much MP as Blood Strike wasn't worthwhile. We had a choice between raising Cleave's damage and lowering its MP cost and chose MP cost. We wouldn't be opposed to making the damage change instead, though personally I suspect that if we do that, it would be best to make Blood Strike also use more MP in exchange for more damage.

The only other MP cost reductions are Flame and Thunderstrike. Those reductions were just because we felt that costing 2 times as much as the respective bolt spell felt more appropriate than costing 2.3–3 times as much. If you find after your testing that you disagree, though, that's fine with us.

🕷️

dorkster commented:
From my testing so far, I think I'm perfectly fine with the lowered MP costs.

- Warrior skills feel well balanced, no further changes there.
- Ranger skills are good, except that Bear Trap feels over-powered. I think the stun/immobilize time should be more on par with the Magician's Thunderstrike skill.
- Magician skills seem to be okay, but I need to spend more time play testing those.

onpon4 commented:
Regarding the Bear Trap's power, I think lowering its stun/immobilize to Thunderstrike level (which is only 1–2 seconds) would make it far too weak, so how about, say, 2–10 seconds or 4–12 seconds instead (making it a bit longer than caltrops, which slow for 2–6 seconds)?

What my testing showed for caltrops and bear traps is that the bear trap has an inherent disadvantage in every setting that isn't a boss fight, since it only hits at most one target in addition to needing the setup (while the redesigned caltrops are multi-hit, meaning they work pretty much everywhere). I think therefore it's important to make the Bear Trap skill have a quite strong effect. That said, we picked the duration before coming up with the idea of making the bear trap stun and you're probably right that stunning/immobilizing for 3 times as long as caltrops is over-the-top.

🕷️

dorkster commented:
The time-to-kill against most enemies is under 3-4 seconds, so I feel like anything higher than that for a stun duration is excessive. Add on top of that the extra damage, stun effect (neutralizing ranged enemies), and the reduced MP cost. The Scout/Ranger is already a strong class (safe from melee, high crit rate, etc), and disabling an enemy like Bear Trap does practically makes the player invincible.

So I'd leave the effect duration as a flat value. Either the original 3 seconds, or if you really want a buff, 4 seconds. That also would allow us to keep a single post power for the visual effect. The current post power lingers for too long in your version when the power is not level 5. Though that would be fixable by adding additional post powers for each level.

----

I've also play tested more with further power investment. The damage increases for powers that can hit multiple enemies are too high. I balanced them under the assumption that the player would be hitting at least 2 enemies at a time with them. So the player's total damage output is 200% if the power's base damage is 100%. Considering these powers come out as fast as ones that damage a single enemy, they already had an advantage in terms of rate of damage over time.

Some of the single-target power damages might also be too high, but I would start with those multi-target ones (Burn, Freeze, etc). Those ones were the most drastically affected.

----

I think we may have some disagreement in terms of how easy/hard the game should be. What are your thoughts on making this its own mod? I'd probably still want to cherry-pick a few things for upstream (the Caltrops changes immediately come to mind). We can of course keep going to find more middle ground, but I don't want to ruin your idea of "perfect" either.

onpon4 commented:
> The time-to-kill against most enemies is under 3-4 seconds, so I feel like anything higher than that for a stun duration is excessive. Add on top of that the extra damage, stun effect (neutralizing ranged enemies), and the reduced MP cost. The Scout/Ranger is already a strong class (safe from melee, high crit rate, etc), and disabling an enemy like Bear Trap does practically makes the player invincible.
> 
> So I'd leave the effect duration as a flat value. Either the original 3 seconds, or if you really want a buff, 4 seconds. That also would allow us to keep a single post power for the visual effect. The current post power lingers for too long in your version when the power is not level 5. Though that would be fixable by adding additional post powers for each level.

Regarding the bear trap: what I'm most concerned about with just leaving it a flat, modest immobilize period is ensuring that those power upgrades for it are still meaningful. Upgrading a power is expensive (you only get 15, maybe 16 skill points in a given play-through). Something we've consistently been doing here is trying to make upgrading a power to level 5 close to as good as getting 5 separate powers. In the case of bear traps, what felt most natural to us to do that was to simply increase its trap duration, then when we found that wasn't enough, we made it stun enemies as well.

We designed the traps to handle swarms of enemies, not single enemies. In that context, the very long duration of the bear trap's stun/immobilize can keep some enemies at bay while you take care of others. I don't think the effect is that strong when it only lasts 3 seconds; sure, that's how much time it takes to kill a single enemy, but if you're fighting a cluster of, say, 10 enemies, those 3 seconds just don't affect the outcome anywhere near as much as the damage the bear trap does.

That context also means that the bear trap has an inherent disadvantage as compared to the upgraded caltrops, given that it can only affect a single target, as Layla mentions. The addition of stunning helps, but how much that helps decreases as the duration decreases. Assuming it takes 2 seconds to kill each enemy, stunning a boss for 12 seconds (for instance) gives you enough time to kill 6 small enemies, but by contrast, stunning the boss for only 3 seconds barely gives you enough time to kill one.

I would also like to note that we originally thought the stun period would cause a serious problem with bosses, but since the stun effect ends the moment you hit the enemy with another attack, the only bosses this makes you invincible against are early game melee bosses. For other bosses, the only way to take advantage of the stun effect lasting a long time is to kill helper enemies. For example, you can stun Mez with the bear trap and run off and kill the wyverns in the arena without having to worry about Mez interfering for awhile, but if you were to bear-trap Mez and then attack him, the second one of your arrows hits him, he can start using his lightning on you, though he won't be able to move thanks to the immobilize effect.

I agree that 5–18 seconds is probably overkill, but I think that for the stun time in particular, it just wouldn't do much if it was locked to 2–3 seconds. Maybe we could do a hybrid approach: an immobilize period which is a static number like 3 seconds (that's the original immobilize time), but a stun period that lengthens. Since the stun effect has that inherent disadvantage of auto-disabling when you attack the enemy, this would mean that you still get the kind of benefit I mentioned above for dealing with swarms, but it wouldn't oversimplify melee and short-range bosses with the immobilize duration. Thoughts?

Alongside this, we could also reduce the amount of damage increase from the traps, since the damage they inflict is really not their primary benefit. How would you feel about 60–80 damage for caltrops and 120–140 damage for the bear trap?

> I've also play tested more with further power investment. The damage increases for powers that can hit multiple enemies are too high. I balanced them under the assumption that the player would be hitting at least 2 enemies at a time with them. So the player's total damage output is 200% if the power's base damage is 100%. Considering these powers come out as fast as ones that damage a single enemy, they already had an advantage in terms of rate of damage over time.

I suppose I can see the possibility that Burn and Thunderstrike should do the same damage as Fireball and Shock based on your reasoning. I'll implement that change. I'm not really convinced this applies to Freeze, though, simply because of how difficult Freeze is to use (and how nearly impossible it is to use offensively). I'd definitely be interested in making a change to fix that, though, so that its short range and lack of flexibility are compensated for. Freeze is the one magician skill that I've always felt is useless in my playthroughs, even though it's one of my favorites visually.

Right now, all Freeze is really good for is as a defensive move. In some ways this makes it not unlike Quake, but the effect of slowing enemies down (even if the slowdown is a lot) really just doesn't contribute much. Making it faster would probably help, but I suspect this would just render the ice bolt skill redundant. Making it a directed skill like Thunderbolt and Burn would certainly work, but I feel that would make the Freeze skill lose its identity. There must be some other thing that can be done, like an additional status effect or something.

> I think we may have some disagreement in terms of how easy/hard the game should be.

Well, no, this isn't really about how easy or hard the game is. It's more to do with how worthwhile power upgrades are. That's the focus of this PR: ensuring that it isn't a mistake for the player to upgrade a power.

To demonstrate what I mean, I want to point to what I think is the worst example, the fireball skill. In the current Empyrean Campaign release, level 1 fireball inflicts 105% damage and costs 2 MP, while level 2 inflicts 110% damage and costs 3 MP. That means that if you spend a skill point to level up the fireball skill, you get a relative damage increase of about 4.8%, but a relative MP cost increase of 50%. This means the skill will kill enemies 4.8% faster, but at a cost of draining your MP 50% faster. Considering you're not really going to notice enemies dying 4.8% faster (it might not even have an effect depending on exact health values), but you *are* going to notice your MP draining 50% faster, this is not a worthwhile trade-off.

So what we did to the fireball is make level 1 inflict 130% damage and make level 2 inflict 160% damage (for the same respective MP costs). That means for that 50% increase in MP cost, you get a 23% increase in damage output. You'll find this is consistent throughout our changes: we strove to make it so that powers would have around half as much of a damage increase as the corresponding MP increase. This makes the upgrade still less MP efficient, but at least makes the increased killing speed worth the trade-off.

I don't find the Empyrean Campaign hard at all, but I'm not opposed to a hard game. Actually, I half-considered including buffs to the enemy Freeze skill, but decided that doesn't really fit the scope of this PR.

🦇

onpon4 commented:
I've gone ahead and made the power increases of Burn and Thunderstrike more in-line with Fireball and Shock, and implemented that change to bear traps and caltrops that I suggested. Please let me know what you think!

🦇

dorkster commented:
Better, but shouldn't the effects on the Bear Trap be switched? Stun also immobilizes the target, so a stun of 10s means no attacking OR moving for 10s. I think it should be a 3s flat stun, with incrementing immobilization. That way, an enemy will recover from the initial stun (and can thus attack), but still be prevented from moving for longer.

onpon4 commented:
> Better, but shouldn't the effects on the Bear Trap be switched? Stun also immobilizes the target, so a stun of 10s means no attacking OR moving for 10s (provided they aren't knocked out of the stun by an attack). I think it should be a 3s flat stun, with incrementing immobilization. That way, an enemy will recover from the initial stun (and can thus attack), but still be prevented from moving for longer.

Well we think no, because it's the stun that's more important and at the same time it's the immobilize effect that allows cheesing early bosses. Like Eris explained, it's because the stun effect stops the moment they get damaged again. So unless you're holding them in place to stop you from interfering with something else you're doing (e.g. killing different enemies), the stun effect doesn't really do much. Conversely, though, the immobilize effect without stun doesn't really help with any of the dangerous enemies past the first boss, so a 10 second immobilize isn't really very useful (in fact even that 18 second immobilize we found to be almost useless before we implemented the stun effect alongside it).

So this is the idea we implemented here:

* The stun effect is sort of the signature "big deal" of the bear trap after the early game, and is the main reason you would want to upgrade the bear trap.
* The immobilize effect's main purpose is to have indiscriminate uses of the bear trap always does something useful for at least 3 seconds, to limit the mitigation of its effect that happens when you hit a trapped enemy.

We're generally not trap players, so we won't be too disappointed if you would prefer to do it the way you suggested, but that's is the reasoning.

🕷️

dorkster commented:
> Well we think no, because it's the stun that's more important and at the same time it's the immobilize effect that allows cheesing early bosses.

I disagree. The primary purpose of the traps is to help Rangers create spacing between themselves and enemies. So controlling the target's movement is the most important thing. Also, it's likely that the player will attack (and knock out of stun) a trapped enemy soon after having trapped them. The stun effect would just be an extra little bonus against ranged enemies if the player is using the bear trap to do crowd control. *Note*: The Bear Trap should be better vs a single enemy, whereas Caltrops should be better for crowd control.

I also think it's worth illustrating that it takes a little over 3 seconds to escape a non-moving enemy's threat range when the player is moving at the default speed (20 tiles / 6 tiles-per-second). That's why 18, and even 10, seconds is way over the top in my opinion. I think 5 seconds would be more appropriate.

onpon4 commented:
OK, so to clarify, you want a static 5 second combined stun and immobilize for the bear trap? I'm fine with that if that's how you want it.

🕷️

dorkster commented:
Good, but I would still reduce the stun time to something like 1 second. An initial "shock" to the enemy that they've been trapped, but let them return to their normal state. Looks weird to have the enemy completely frozen for 5 seconds.

onpon4 commented:
Alrighty, I made that change. Let us know if any other changes needed! 🙂

🕷️

onpon4 commented:
Ah, lemme also brainstorm ways to make Freeze strong enough to justify a high cost like the other multi-hits without just making it do ∞+1 damage:

* Maybe some sort of weakness thingy like the spell some of those elemental skeletons cast? Probably not, doesn't seem like it fits the short-rangeness of it.
* Maybe it might be possible to have it block projectiles? Though that wouldn't solve the problem of making upgrades to it worthwhile.
* Could be, like, slow but long-range, maybe? Like maybe it just lasts longer and you'd be able to shoot one and sort of follow it, or you could throw it at an horde of enemies that's approaching you and be able to rely on it to help you to get away.

Any thoughts?

🕷️

onpon4 commented:
OK, so that slow but long-range idea, I decided to give it a test run, slowed it down 25% but increased the range (doubled "count", but that's not quite a doubled range because of the slower speed). I think this works, tried it out with increase MP cost and lowered damage (matching Burn/Thunderstrike MP cost and ice bolt damage) and it seems to work well, especially in corridors. Lemme know what you think!

🕷️

onpon4 commented:
Hi, any more thoughts about the suitability of our changes here, @dorkster? Is there anything else that needs to be changed before this can be merged?

🦇

dorkster commented:
I've actually been working on a set of balance changes that will conflict with this PR. The purpose being:

1. To allow for more mixed and varied builds, using powers across trees
2. More importantly, to lessen the ways that a player can make a "bad" build that makes it hard to progress.

My solution is:

- [x] Change the weight of how player leveling / stat investment increases damage versus weapon / item bonuses. What I've done is make it so that what would have been a +5 item bonus per level becomes a +3/+2 split between player stats and item bonus. This benefits mixed builds because it lets the player do some damage with powers they might not specialize in.
- [x] Award 2 power points per level instead of just 1. It wasn't even possible to max a single class tree with the current leveling scheme. Doubling the points give players more freedom to experiment, and also makes investing a point less dramatic.
- [ ] Level 1 of each power should be significantly strong, with a less steep upgrade slope. This may seem to run counter to the idea of making the leveling up of powers more rewarding. But remember, we're giving the player double the power points, so they can invest quickly in a single upgrade branch if they so choose. 
- [ ] MP usage should possibly be flat or a shallow curve across upgrades. Both MP usage and the player's MP pool should be increased. That way, a +1 increase to MP usage for a power isn't as drastic (and I don't want to have fractional MP costs).
- [ ] Lower stat requirements for power upgrades to accommodate the double power points
- [x] Your idea: "_Powers that require other powers as prerequisite only require the first level and then can be upgraded without upgrading the prerequisite power_". This is good, and I'm essentially cherry-picking it (although I rewrote it so it'd be in a single commit).

Outside of that last point, there's not much I can use directly. But your observations are valuable, and I've been using this PR as a reference for which powers need more attention. There's also a few non-numerical changes, such as making Caltrops multi-hit, that I'll be adapting. So, where I wouldn't be merging this PR, it makes sense to me to credit you as a contributor on my set of changes. Does that sound reasonable?

onpon4 commented:
Well it's not my game, so ultimately you have the final say in what goes into it. I'll say what I think about your proposal, thô, for what it's worth:

* Enabling more mixed builds sounds like a nice idea, and I'll be looking forward to experiencing that. (I do rather like the idea of promoting more mixed builds.)
* Doubled points is also something I can get behind, thô I don't see it as really necessary.
* As far as keeping the paradigm of very weak upgrades while increasing points and lowering MP cost increases, I do think that would be a slight improvement over the current paradigm where upgrades are arguably actually detrimental, but I would caution that minor upgrades for the cost of a whole new power that is much more significant will still leave upgrades of powers very undesirable, especially since you're also adding more capability for diversification. You didn't mention numbers here, but if, for example, upgrading fireballs to level 2 only gave a 5% boost to damage, it would still be the absolute last thing I would invest points in. If you *want* the meta-game to be ignoring those upgrades, I obviously can't stop you, but I think you should be cognizant of the fact that making the upgrades minor is going to do that.
* The one thing you said you pulled from this is honestly a rather minor improvement, but it's still welcome. I especially think it's unimportant because you say you're going to make upgrades still a minor thing.

I suspect, overall, that your proposed changes won't impact how I tend to decide on builds. As is the case in the current version of the Empyrean Campaign, I would just put 1 point into all of the powers I need and invest the rest into the stat boosts, utilities like Warcry, Haste, and Shield, and then just one each of the powers I don't need. Yes, having 32 points available and not increasing MP costs much does make it more feasible to, for example, upgrade the fireball skill to level 5, but for the cost of a minor damage boost I'm not going to see, it's just a better move to invest those same points into five whole skills, even if there's very little chance I'll ever use them. The upgrade is also still just as much of a "trap", since performing the upgrade leads to a lack of diversification which isn't being made up for by the tiny damage increase, and the fact that you can overcome that in the late game won't do anything to solve that problem in the early game.

🦇

dorkster commented:
>but I would caution that minor upgrades for the cost of a whole new power that is much more significant will still leave upgrades of powers very undesirable, especially since you're also adding more capability for diversification. You didn't mention numbers here, but if, for example, upgrading fireballs to level 2 only gave a 5% boost to damage, it would still be the absolute last thing I would invest points in.

I understand. My point is more about the jump in power at level 1 versus the jump for each upgrade. For example, maybe Fireball starts at 160% damage and then gets 20% per level. The initial jump to +60% damage is greater than the per level bonus, but the per level bonus is still significant enough to matter. Don't get too hung up on the exact numbers, because I'm unsure myself. I think the idea is to make the upgrades *just* attractive enough when placed beside new powers, but no more.

>but for the cost of a minor damage boost I'm not going to see, it's just a better move to invest those same points into five whole skills, even if there's very little chance I'll ever use them.

My goal is to even out how valuable things are. Currently, the "optimal" build is to max: 1 damage power, (or 2 if one is multi-hit), the 3 passives, and possibly Shield/Haste. So investing in multiple powers instead of a single one puts the player at a _disadvantage_. My proposed changes should make it so that the player doesn't completely neuter themselves if they decide that the upgrades aren't worth it.

Two power points per level means the player can unlock every level 1 skill by the time they reach level 12. Considering the game is designed for the player to be around level 16 by end-game, that means they'd still be able to max out 2 powers in their playthrough. Most players aren't going to want to unlock _every_ power, so I think it's more likely that they'll get the build they want without having to wait until end-game.

onpon4 commented:
Well, going from 160% to 180% would be a 12.5% upgrade in damage, so that would still be much better than current 4.8% damage upgrade in the release. I'll be looking forward to that. :+1:

I'm not sure what you mean when you say that "currently" investing in multiple powers puts the player at a disadvantage. Is that in reference to this PR, or to the latest release of the Empyrean Campaign? The statement makes sense if in reference to this PR, but if in reference to the current Empyrean Campaign release, I'm not sure why you would say that; it runs counter to my experience playing the game, particularly in the mage tree (where, most egregiously, level 5 fireball inflicts only 19% more damage than level 1 fireball, but costs three times as much in MP).

🦇

dorkster commented:
>I'm not sure what you mean when you say that "currently" investing in multiple powers puts the player at a disadvantage. Is that in reference to this PR, or to the latest release of the Empyrean Campaign?

I was referring to the latest (1.14) version of the Empyrean Campaign. Provided the player has enough MP, there's no real reason to use a lower level power when one is at level 5. The MP bonuses from player levels and items is designed to accommodate the increased cost of power upgrades.

It sounds like your gripe is less with the amount of bonus damage, but rather the ratio between the damage and the MP usage. I can see why that would make it seem "not worth it". Question: How would you feel if level 5 was still 19% more damage than level 1, but it had the same (or close to) cost as level 1? Anyway, I think flattening the MP inflation would make things better here.

onpon4 commented:
Well, it's not just the ratio, althô that does figure into it. It's also just that a 19% damage boost is wholly insignificant compared to the extra health afforded by the shield power, or just the stat boosts. The same goes for the boosts in the other powers; I just find the fire powers to be the most egregious in this regard.

I've actually attempted to run a fire-only build in the past (in a previous release of the Empyrean Campaign), and performance with that was so abysmal that I never finished that game. You lose access to the ability to take advantage of elemental weaknesses and being able to defensively use the ice slowdown on fast enemies, and the slight damage boosts didn't even come close to making up for that. The increase in MP cost was a major problem as well, but even if there was zero MP cost increase to the upgrades, I wouldn't consider a 4.8% damage boost to be anywhere near as valuable as any of the stat boosts, the shield power, the Warcry power, or the Haste power. It would be one of those things in a skill tree I would only get if there's nothing at all of interest, and that's assuming zero increase in cost.

Keep in mind, a 19% increase in damage is very negligible in practice because any extra damage just doesn't really exist if it wasn't needed to defeat an enemy. Just taking some made-up numbers as an example, if an enemy has 300 HP and the level 1 fireball inflicts 200 damage, I would need two fireballs to defeat the enemy regardless of whether it's at level 1 or level 5. The only way that 19% damage boost is going to matter is if the enemy's HP is between 201 and 238. (Of course, in the actual game the numbers get modified more dynamically than that, but the point is that tiny damage increases just aren't going to make an exactly proportional difference against most enemies, and in general the benefit is going to be far less than the raw numbers might predict.)

🦇

dorkster commented:
I agree that the 4.8% boost is too little. I was just trying to say that the damage might not need to be increased as much if the MP cost is kept under control.

Thanks for all the feedback. I definitely have some more direction for balancing now.

dorkster commented:
Sorry it took so long to get back to this. I just pushed my set of balance changes:

- b58962250fe1a1d9b3ab327d1f2a3e517fcfd229
- e8edd3b18935497892cda4662d427d21281b3ce9
- e1a3da84a1ef4abf8543cbd80cf77ffcf7ec09b8
- dfbd34fac8a8148cb705e42f7a9b3295dca214a4
- 39cccf7ac14ed6b2848a2448bde6855c992ed1e3
- 3b1e2823228bad0b723099879f0b3266f7f2ff72
- f4bfb217290e0a95927080a7ec2e576cbb68c4ec
- 9530f29322da0d1c401700dad0575126f0f312dd
- 504a55028f122c8751b38b3162db20da4dc100eb

While different in some regards, it should be closer to what you had in this PR. I'm going to close this, since any future balance tweaks will be done based on this new baseline.
